url/IdnaTestV2.window.js

```diff
@@ -0,0 +1,41 @@
+promise_test(() => fetch("resources/IdnaTestV2.json").then(res => res.json()).then(runTests), "Loading data…");
+
+// Performance impact of this seems negligible (performance.now() diff in WebKit went from 48 to 52)
+// and there was a preference to let more non-ASCII hit the parser.
+function encodeHostEndingCodePoints(input) {
+  let output = "";
+  for (const codePoint of input) {
+    if ([":", "/", "?", "#", "\\"].includes(codePoint)) {
+      output += encodeURIComponent(codePoint);
+    } else {
+      output += codePoint;
+    }
+  }
+  return output;
+}
```

Comment:
It would be slightly nicer for other consumers like jsdom if this preprocessing, as well as the prepending of `https://` and appending of `/x`, was done by the Python script that output the JSON file.

Reply:
I did not want to do that as I want `URLHost` (it'll happen one day!) to be able to consume the actual test.